src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -34,6 +29,7 @@ public interface IStepHost : IRunnerService
                                bool killProcessOnCancel,
                                bool inheritConsoleHandler,
                                CancellationToken cancellationToken);
+        DictionaryContextData GetExpressionValues(IExecutionContext executionContext);
```

Comment:
add an empty line before this.

Reply:
seems weird that we add an interface method that will eventually just call `ResolvePathForStepHost()`

Reply:
Thanks for the review!  

My thoughts are that the logic is only dependent on the execution context and the step host. Since the IExecutionContext is being used in a lot of places, and StepHost does the translation, it is clear when we call it to see that we want expression values for that StepHost. We pass the execution context, and we get expression values that should be used when the template is evaluated. That way, wherever we use template evaluation and we have a step host, we just call the step host to translate those values for the template.

Also, if the new StepHost is introduced, it should also know what expressionValues should be for it to work properly

Reply:
I'm inclined to agree the logic in `GetExpressionValues` and private method `UpdatePathsInExpressionValues` belongs in `ActionRunner.cs`, the translation of ExpressionValues like this feels like a general case. 

In StepHost, we eventually do just call `ResolvePathForStepHost`. Most of this method is prepwork for `Container.TranslateToContainerPath`. This 'prepwork' is not operating on any properties of 'StepHost' other than 'Container', so it doesn't present a problem to cross reference from `GetExpressionValues` to `ResolvePathForStepHost`.

Alternatively, we could modify ResolvePathForStepHost and move all the the logic between `ArgUtil.NotNull(Container, nameof(Container));` and `Container.MountVolumes.Exists` to a private method `isPathContainerTranslatable`. `isPathContainerTranslatable` can then be shared between ResovePathForStepHost and GetExpressionValues?